xarray/core/groupby.py

```diff
@@ -45,17 +46,23 @@
     peek_at,
 )
 from xarray.core.variable import IndexVariable, Variable
+from xarray.namedarray.pycompat import is_chunked_array
 from xarray.util.deprecation_helpers import _deprecate_positional_args
 
 if TYPE_CHECKING:
     from numpy.typing import ArrayLike
 
     from xarray.core.dataarray import DataArray
     from xarray.core.dataset import Dataset
-    from xarray.core.types import GroupIndex, GroupIndices, GroupKey
+    from xarray.core.types import GroupIndex, GroupIndices, GroupKey, T_Chunks
     from xarray.core.utils import Frozen
     from xarray.groupers import Grouper
 
+    if sys.version_info >= (3, 11):
+        from typing import Self
+    else:
+        from typing_extensions import Self
```

Comment:
We have dealt with this in:
https://github.com/pydata/xarray/blob/3c19231a106ceb147c1b6f9021cdaca817df3138/xarray/core/types.py#L19-L30